test/tools/Modules/WebListener/WebListener.psm1

```diff
@@ -133,48 +137,53 @@ function Start-WebListener
 
-        $Job = Start-Job {
-            $path = Split-Path -parent (get-command WebListener).Path -Verbose
-            Push-Location $path -Verbose
-            'appDLL: {0}' -f $using:appDll
-            'serverPfxPath: {0}' -f $using:serverPfxPath
-            'serverPfxPassword: {0}' -f $using:serverPfxPassword
-            'HttpPort: {0}' -f $using:HttpPort
-            'Https: {0}' -f $using:HttpsPort
-            'Tls11Port: {0}' -f $using:Tls11Port
-            'TlsPort: {0}' -f $using:TlsPort
+        $oldASPEnvPreference = $env:ASPNETCORE_ENVIRONMENT
+
+        try {
             $env:ASPNETCORE_ENVIRONMENT = 'Development'
-            dotnet $using:appDll $using:serverPfxPath $using:serverPfxPassword $using:HttpPort $using:HttpsPort $using:Tls11Port $using:TlsPort
+
+            $params = @{
+                FilePath = $appExe
+                PassThru = $true
+                UseNewEnvironment = $IsLinux # start a non-keyboard input blocking process on linux.
+            }
+
+            $webListenerProcess = Start-Process @params -ArgumentList @(
+                $serverPfxPath
+                $serverPfxPassword
+                $HttpPort
+                $HttpsPort
+                $Tls11Port
+                $TlsPort
+            )
+        } catch {
+            # rethrow any exception
+            throw $_
+        } finally {
+            $env:ASPNETCORE_ENVIRONMENT = $oldASPEnvPreference
         }
+
         $Script:WebListener = [WebListener]@{
             HttpPort  = $HttpPort
             HttpsPort = $HttpsPort
             Tls11Port = $Tls11Port
             TlsPort   = $TlsPort
-            Job       = $Job
+            Process   = $webListenerProcess
         }
 
         # Count iterations of $sleepMilliseconds instead of using system time to work around possible CI VM sleep/delays
         $sleepCountRemaining = $initTimeoutSeconds * 1000 / $sleepMilliseconds
         do
         {
             Start-Sleep -Milliseconds $sleepMilliseconds
-            $initStatus = $Job.ChildJobs[0].Output | Out-String
-            $isRunning = $initStatus -match $initCompleteMessage
+            $isRunning = (Get-WebListener).Process -ne $null
             $sleepCountRemaining--
         }
         while (-not $isRunning -and $sleepCountRemaining -gt 0)
 
         if (-not $isRunning)
         {
-            $jobErrors = $Job.ChildJobs[0].Error | Out-String
-            $jobOutput =  $Job.ChildJobs[0].Output | Out-String
-            $jobVerbose =  $Job.ChildJobs[0].Verbose | Out-String
-            $Job | Stop-Job
-            $Job | Remove-Job -Force
-            $message = 'WebListener did not start before the timeout was reached.{0}Errors:{0}{1}{0}Output:{0}{2}{0}Verbose:{0}{3}' -f
-                ([System.Environment]::NewLine), $jobErrors, $jobOutput, $jobVerbose
-            throw $message
+            throw 'WebListener did not start before the timeout was reached.'
```

Comment:
The original message provides more diagnosis information, which could be useful in case it fails. Maybe we should keep that.